internal/config/utils.go

```diff
@@ -91,3 +94,83 @@ func doViperBind[V any](
 
 	return nil
 }
+
+// GetConfigFileData returns the data from the given configuration file.
+func GetConfigFileData(cfgFile, defaultCfgPath string) (interface{}, error) {
+	var cfgFilePath string
+	var err error
+	if cfgFile != "" {
+		cfgFilePath, err = filepath.Abs(cfgFile)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		cfgFilePath, err = filepath.Abs(defaultCfgPath)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	cleanCfgFilePath := filepath.Clean(cfgFilePath)
+	if info, err := os.Stat(cleanCfgFilePath); err == nil && info.IsDir() || err != nil && os.IsNotExist(err) {
+		return nil, nil
+	}
```

Comment:
While mounting, if no config file is present on local, docker will create an empty mount path in the container, but that would be a directory, so we skip that. 

We ignore if no config file exists, the system will revert to the defaults.

Reply:
(I'm approving without requiring a change, but this would be a great comment in the code itself, since it's not immediately obvious what these two cases are.)

Reply:
> (I'm approving without requiring a change, but this would be a great comment in the code itself, since it's not immediately obvious what these two cases are.)

I'll cover this in next PR.